tests/test_serialize.py

```diff
@@ -123,93 +134,112 @@ def test_tensor_non_tensor_return(self, monkeypatch):
 
         mock_obs = mock.MagicMock()
 
+        use_csingle = True if ObsFunc == ObsStructC64 else False
+        obs_str = "ObsStructC64" if ObsFunc == ObsStructC64 else "ObsStructC128"
+
         with monkeypatch.context() as m:
-            m.setattr(pennylane_lightning._serialize, "ObsStructC128", mock_obs)
-            _serialize_obs(tape, self.wires_dict)
+            m.setattr(pennylane_lightning._serialize, obs_str, mock_obs)
+            _serialize_obs(tape, self.wires_dict, use_csingle=use_csingle)
 
         s = mock_obs.call_args[0]
         s_expected = (
             ["Hermitian", "Hermitian"],
             [np.eye(4).ravel(), np.eye(2).ravel()],
             [[0, 1], [2]],
         )
-        ObsStructC128(*s_expected)
+        ObsFunc(*s_expected)
 
         assert s[0] == s_expected[0]
         assert np.allclose(s[1][0], s_expected[1][0])
         assert np.allclose(s[1][1], s_expected[1][1])
         assert s[2] == s_expected[2]
 
-    def test_mixed_tensor_return(self, monkeypatch):
+    @pytest.mark.parametrize("ObsFunc", [ObsStructC128, ObsStructC64])
+    def test_mixed_tensor_return(self, monkeypatch, ObsFunc):
         """Test expected serialization for a mixture of Hermitian and Pauli return"""
         with qml.tape.QuantumTape() as tape:
             qml.expval(qml.Hermitian(np.eye(4), wires=[0, 1]) @ qml.PauliY(2))
 
         mock_obs = mock.MagicMock()
 
+        use_csingle = True if ObsFunc == ObsStructC64 else False
+        obs_str = "ObsStructC64" if ObsFunc == ObsStructC64 else "ObsStructC128"
+
         with monkeypatch.context() as m:
-            m.setattr(pennylane_lightning._serialize, "ObsStructC128", mock_obs)
-            _serialize_obs(tape, self.wires_dict)
+            m.setattr(pennylane_lightning._serialize, obs_str, mock_obs)
+            _serialize_obs(tape, self.wires_dict, use_csingle=use_csingle)
 
         s = mock_obs.call_args[0]
         s_expected = (["Hermitian", "PauliY"], [np.eye(4).ravel()], [[0, 1], [2]])
-        ObsStructC128(*s_expected)
+        ObsFunc(*s_expected)
 
         assert s[0] == s_expected[0]
         assert np.allclose(s[1][0], s_expected[1][0])
         assert s[2] == s_expected[2]
 
-    def test_integration(self, monkeypatch):
+    @pytest.mark.parametrize("ObsFunc", [ObsStructC128, ObsStructC64])
+    def test_integration(self, monkeypatch, ObsFunc):
         """Test for a comprehensive range of returns"""
         wires_dict = {"a": 0, 1: 1, "b": 2, -1: 3, 3.141: 4, "five": 5, 6: 6, 77: 7, 9: 8}
-        I = np.eye(2).astype(np.complex128)
-        X = qml.PauliX.matrix.astype(np.complex128)
-        Y = qml.PauliY.matrix.astype(np.complex128)
-        Z = qml.PauliZ.matrix.astype(np.complex128)
+        I = np.eye(2).astype(np.complex64)
+        X = qml.PauliX.matrix.astype(np.complex64)
+        Y = qml.PauliY.matrix.astype(np.complex64)
+        Z = qml.PauliZ.matrix.astype(np.complex64)
```

Comment:
Should we have both?

Reply:
Of course! I should have added already :slightly_smiling_face: